### ISSUE TITLE
Out of the box compatibility with surface-based methods

### ISSUE THREAD

**Issue by yyeboah** [KAIR-BAIR/nerfacc#65]
Thanks for sharing this great tool. Very exciting stuff.  

This is more of a question out of curiosity than an issue so please feel free to let me know if it's been asked before.  I notice most of the examples shown are for density-based NeRFs. I wonder if this works out of the box with surface-based methods, such as UniSurf and the numerous other works in that direction. If not, what adaptations need to be made for such compatibility to be achieved?

Thanks for your contributions.

**Comment by liruilong940607** [comment]
Hi this is a very good question!

I think most of the surface-based methods are based on SDF, and has a way to convert SDF to opacity (alpha) values.

We actually have a branch `alpha`, that works with alpha-based NeRFs. My lab-mate @doinksta is currently using that branch to work with NeuS. Though that branch is a bit out dated, the main functionalities are all there.

If you want to use the master branch to work with alpha-based NeRFs, (i.e. say you just want to pip install this repo and use it out of the box), current workaround is to write your own `ray_marching` and `rendering` functions, using all the util functions from nerfacc.

But this is extremely simple, you can just copy paste the `ray_marching` function defined [here](https://github.com/KAIR-BAIR/nerfacc/blob/27b657fa769ed05191eb055a03071a589066a6fc/nerfacc/ray_marching.py#L15), and `rendering` function defined [here](https://github.com/KAIR-BAIR/nerfacc/blob/27b657fa769ed05191eb055a03071a589066a6fc/nerfacc/vol_rendering.py#L15), and do a bit modification:

1.  Modify 
https://github.com/KAIR-BAIR/nerfacc/blob/27b657fa769ed05191eb055a03071a589066a6fc/nerfacc/ray_marching.py#L190-L194
into:
```python
alphas = alpha_fn(t_starts, t_ends, ray_indices.long())
```

2.  Modify 
https://github.com/KAIR-BAIR/nerfacc/blob/27b657fa769ed05191eb055a03071a589066a6fc/nerfacc/vol_rendering.py#L92-L94
into:
```python
weights = render_weight_from_alpha( 
     packed_info, alphas, early_stop_eps, alpha_thre 
 ) 
```

3. replace `sigma_fn` to `alpha_fn`, and replace `rgb_sigma_fn` with `rgb_alpha_fn`.

Basically, now instead of defining a `sigma_fn` with your NeRF, you should define an `alpha_fn` with it. And I think it should work with most of the SDF based methods.

We plan to support this in the next release!

**Comment by yyeboah** [comment]
@liruilong940607 Thanks for the pointers. It does indeed appear that this should be able to offer support in a trivial manner. I'll give it a go, and looking forward to the stable support version in the next release! Cheers.

**Comment by liruilong940607** [comment]
This is now supported in `nerfacc=0.2.2`! Related PR: #94

**Comment by liruilong940607** [comment]
An neuS example of using nerfacc can be found here: https://github.com/bennyguo/instant-nsr-pl
